cmd/xl-v1-multipart.go

```diff
@@ -416,6 +416,19 @@ func (xl xlObjects) PutObjectPart(bucket, object, uploadID string, partID int, s
 	// Delete the temporary object part. If PutObjectPart succeeds there would be nothing to delete.
 	defer xl.deleteObject(minioMetaBucket, tmpPartPath)
 
+	if size > 0 {
+		for _, disk := range onlineDisks {
+			if disk != nil {
+				sizeInDisk := size / int64(xlMeta.Erasure.DataBlocks)
```

Comment:
> But instead of having the above complicated calculation we can take an approximate value of :
> sizeInDisk := size / int64(xlMeta.Erasure.DataBlocks)

not sure about this one.. the file still could be fragmented if we allocate with a size inferior to the last final size.